pinot-hadoop-filesystem/src/main/java/com/linkedin/pinot/storage/HadoopPinotFS.java

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.storage;
+
+import com.google.common.base.Strings;
+import com.linkedin.pinot.common.segment.fetcher.HdfsSegmentFetcher;
+import java.io.IOException;
+import java.net.URI;
+import java.util.ArrayList;
+import org.apache.commons.configuration.Configuration;
+import org.apache.hadoop.fs.FileUtil;
+import org.apache.hadoop.fs.LocatedFileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RemoteIterator;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.linkedin.pinot.common.utils.CommonConstants.SegmentFetcher.*;
+import static com.linkedin.pinot.common.utils.CommonConstants.SegmentFetcher.HdfsSegmentFetcher.*;
+
+
+/**
+ * Implementation of PinotFS for the Hadoop Filesystem
+ */
+public class HadoopPinotFS extends PinotFS {
+  private static final Logger LOGGER = LoggerFactory.getLogger(HdfsSegmentFetcher.class);
+  private org.apache.hadoop.fs.FileSystem hadoopFS = null;
+  private int retryCount = RETRY_DEFAULT;
+  private int retryWaitMs = RETRY_WAITIME_MS_DEFAULT;
+  private org.apache.hadoop.conf.Configuration hadoopConf;
+
+  public HadoopPinotFS() {
+
+  }
+
+  @Override
+  public void init(Configuration config) {
+    try {
+      retryCount = config.getInt(RETRY, retryCount);
+      retryWaitMs = config.getInt(RETRY_WAITIME_MS, retryWaitMs);
+      hadoopConf = getConf(config.getString(HADOOP_CONF_PATH));
+      authenticate(hadoopConf, config);
+      hadoopFS = org.apache.hadoop.fs.FileSystem.get(hadoopConf);
+      LOGGER.info("successfully initialized hdfs segment fetcher");
+    } catch (Exception e) {
+      LOGGER.error("failed to initialized the hdfs segment fetcher", e);
+    }
+  }
+
+  @Override
+  public boolean delete(URI segmentUri) throws IOException {
+    return hadoopFS.delete(new Path(segmentUri), true);
+  }
+
+  @Override
+  public boolean move(URI srcUri, URI dstUri) throws IOException {
+    return hadoopFS.rename(new Path(srcUri), new Path(dstUri));
+  }
+
+  /**
+   * Note that this method copies within a cluster. If you want to copy outside the cluster, you will
+   * need to create a new configuration and filesystem.
+   */
+  @Override
+  public boolean copy(URI srcUri, URI dstUri) throws IOException {
```

Comment:
Will this throw an error for cross cluster copies?

Reply:
yes